control-plane-operator/controllers/hostedcontrolplane/kas/deployment.go

```diff
@@ -319,6 +325,21 @@ func ReconcileKubeAPIServerDeployment(deployment *appsv1.Deployment,
 			if err := applyKMSConfig(&deployment.Spec.Template.Spec, secretEncryptionData, images); err != nil {
 				return err
 			}
+
+			// Add the adapter-init and adapter-server containers for ARO HCP
+			if azureutil.IsAroHcp() {
+				deployment.Spec.Template.Spec.InitContainers = append(deployment.Spec.Template.Spec.InitContainers, azureutil.AdapterInitContainer())
+
+				azureCredentials, err := azureutil.GetAzureCredentialsFromSecret(ctx, c, hcp.Namespace, hcp.Spec.Platform.Azure.Credentials.Name)
+				if err != nil {
+					return err
+				}
+
+				deployment.Spec.Template.Spec.Containers = append(deployment.Spec.Template.Spec.Containers, azureutil.AdapterServerContainer(string(azureCredentials.Data["AZURE_CLIENT_ID"]), string(azureCredentials.Data["AZURE_CLIENT_SECRET"]), string(azureCredentials.Data["AZURE_TENANT_ID"])))
```

Comment:
where's this AdapterServerContainer contract defined?

Reply:
https://redhat-external.slack.com/archives/C075PHEFZKQ/p1725584576709039